src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodDriver.cs

```diff
@@ -0,0 +1,190 @@
+                default:
+                    throw new ArgumentException($"Invalid Gpio pin value {value}");
+            }
+
+            return result;
+        }
+
+        protected internal override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void SetPinMode(int pinNumber, PinMode mode)
+        {
+            if (mode == PinMode.InputPullDown || mode == PinMode.InputPullUp)
+            {
+                throw new PlatformNotSupportedException("This driver is generic so it does not support Input Pull Down or Input Pull Up modes.");
+            }
+            _pinNumperToPinModes[pinNumber] = mode;
+        }
+
+        protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventType, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();  // TODO
+        }
+
+        protected internal override void Write(int pinNumber, PinValue value)
+        {
+            string stringValue = ConvertPinValueToSysFs(value);
+            string command = $"--mode=signal {GpioChip0} {pinNumber}={stringValue}";
+            try
+            {
+                Process myProc;
+                if (PinValue.Low == value)
+                {
+                    if (_pinNumperToProcess.ContainsKey(pinNumber))
+                    {
+                        myProc = _pinNumperToProcess[pinNumber];
+                        myProc.Kill();
+                        myProc.WaitForExit((int)TimeSpan.FromMilliseconds(1).TotalMilliseconds);
+                        HandlProcessExit(myProc, $"{GpioSet} {command}", 1);
+                        _pinNumperToProcess.Remove(pinNumber);
+                    }
+                }
+                else
+                {
+                    myProc = new Process();
+                    myProc.StartInfo.FileName = GpioSet;
+                    myProc.StartInfo.Arguments = command;
+                    myProc.Start();
+                    _pinNumperToProcess[pinNumber] = myProc;
+                }
+            }
+            catch (UnauthorizedAccessException e)
+            {
+                    throw new UnauthorizedAccessException("Reading a pin value requires root permissions.", e);
+            }
+        }
+
+        private static bool HandlProcessExit(Process myProc, string command, int waitMilliseconds)
+        {
+            myProc.WaitForExit(waitMilliseconds);
+            if (myProc.HasExited)
+            {
+                if (myProc.ExitCode != 0)
+                {
+                    throw new Exception($"Unknown error occured while running command: {command}");
+                }
+            }
+            else
+            {
+                throw new Exception($"Unable to finish process, timed out while running command: {command}");
+            }
+            return true;
```

Comment:
HandleProcessExit shouldn't return a Boolean, it is not clear to me from the name of the method what does true or false mean in this case. Perhaps you intended something like: `HasProcessExited`?

Reply:
Also, it doesn't look like you are returning false since you are throwing an exception.

Reply:
yeah not returning false in case throwing exception, returning true is for read operation so that if process finished successfully calling function could read the returned value from standard output `string valueContents = gpioGetCommand.StandardOutput.ReadToEnd();`